packages/nhost-js/src/clients/nhost.ts

```diff
@@ -90,4 +90,20 @@ export class NhostClient {
     // this.functions.setAdminSecret(newValue)
     // this.graphql.setAdminSecret(newValue)
   }
+
+  /**
+   * Use `nhost.setRole` to set the user role for all subsequent graphql, storage and functions calls
+   *
+   * @example
+   * ```ts
+   * nhost.setRole('admin')
+   * ```
+   *
+   * @docs https://docs.nhost.io/reference/javascript/set-role
+   */
+  setRole(role: string) {
+    this.graphql.setHeaders({ 'x-hasura-role': role })
```

Comment:
what happens if the user does the following?

```
n.setRole("my-role")
n.storage.setHeaders({"x-something": "blah"}) 
```

will the role still be sent? It won't, will it? We should :)

Reply:
Yes, good catch 👍 - will adjust